### PR TITLE
fix: Multiple runtime fixes/improvments

### PR DESCRIPTION
**Pull request by WojciechMazur**
* Fix order of runtime.Array.copy check for len=0
* Use java.lang.StringBuilder instead of collection.mutable.StringBuilder whenerver possible
* Remove Mac version check in time_nano.c and time.c - fails in MacOS 14 ARM GithubActions runners

### CODE REVIEW

File: javalib/src/main/scala/java/io/DataInputStream.scala [hunk 1]
```diff
@@ -124,7 +124,7 @@ class DataInputStream(in: InputStream)
     var v = in.read()
     if (v == -1) null
     else {
-      val builder = new StringBuilder
+      val builder = new java.lang.StringBuilder
```
Comment (ekrich) [inline javalib/src/main/scala/java/io/DataInputStream.scala:127]: I always just alias `java.lang` -> `jl` and `java.util` -> `ju`. I think that is pretty nice unless it is adding extra symbols or something.

File: nativelib/src/main/scala/scala/scalanative/regex/MachineInput.scala [hunk 2]
```diff
@@ -192,7 +192,9 @@ object MachineInput {
       hayStack match {
         case hayStack: String =>
           hayStack.indexOf(needle, pos)
-        case hayStack: StringBuilder =>
+        case hayStack: java.lang.StringBuilder =>
+          hayStack.indexOf(needle, pos)
+        case hayStack: collection.mutable.StringBuilder =>
```
Comment (ekrich) [inline nativelib/src/main/scala/scala/scalanative/regex/MachineInput.scala:197]: You want both here?

**Review by ekrich — COMMENTED** [review]
Interesting that the macro check doesn't work anymore.

File: nativelib/src/main/scala/scala/scalanative/regex/MachineInput.scala (hunk 2 above)
Comment (WojciechMazur) [reply nativelib/src/main/scala/scala/scalanative/regex/MachineInput.scala:197]: Yes, I don't want to change previous semantics, but rather adapt it to using java.lang.StringBuilder by default

**Pull request merged**